sdk/src/asset_handlers/jpeg_io.rs

```diff
@@ -571,6 +581,257 @@ impl RemoteRefEmbed for JpegIO {
+
+        match dimg {
+            DynImage::Jpeg(jpeg) => {
+                for seg in jpeg.segments() {
+                    match seg.marker() {
+                        markers::APP11 => {
+                            // JUMBF marker
+                            let raw_bytes = seg.contents();
+
+                            if raw_bytes.len() > 16 {
+                                // we need at least 16 bytes in each segment for CAI
+                                let mut raw_vec = raw_bytes.to_vec();
+                                let _ci = raw_vec.as_mut_slice()[0..2].to_vec();
+                                let en = raw_vec.as_mut_slice()[2..4].to_vec();
+
+                                let is_cai_continuation = vec_compare(&cai_en, &en);
+
+                                if cai_seg_cnt > 0 && is_cai_continuation {
+                                    cai_seg_cnt += 1;
+
+                                    let c2pa_bm = box_maps.get_mut(c2pa_bm_index).ok_or(
+                                        Error::InvalidAsset("Invalid C2PA segment".to_owned()),
+                                    )?;
+
+                                    // update c2pa box map
+                                    c2pa_bm.range_len += seg.len_with_entropy();
+                                } else {
+                                    // check if this is a CAI JUMBF block
+                                    let jumb_type = raw_vec.as_mut_slice()[24..28].to_vec();
+                                    let is_cai = vec_compare(&C2PA_MARKER, &jumb_type);
+                                    if is_cai {
+                                        cai_seg_cnt = 1;
+                                        cai_en = en.clone(); // store the identifier
+
+                                        let c2pa_bm = BoxMap {
+                                            names: vec![C2PA_BOXHASH.to_string()],
+                                            alg: None,
+                                            hash: ByteBuf::from(Vec::new()),
+                                            pad: ByteBuf::from(Vec::new()),
+                                            range_start: curr_offset,
+                                            range_len: seg.len_with_entropy(),
+                                        };
+
+                                        box_maps.push(c2pa_bm);
+
+                                        c2pa_bm_index = box_maps.len() - 1;
+                                    } else {
+                                        let name = segment_names.get(&seg.marker()).ok_or(
+                                            Error::InvalidAsset(
+                                                "Unknown segment marker".to_owned(),
+                                            ),
+                                        )?;
+
+                                        let bm = BoxMap {
+                                            names: vec![name.to_string()],
```

Comment:
It looks like this is producing a separate box for every segment while the examples show the boxes gathered into a list of names all hashed as one block. Should we be collecting contiguous segments?

Reply:
We build the target as flat list as we find them.  Note this list does not yet contain hashes.  If we decide we want to hash we allow the caller to decide if the want the collapsed form.